### PR TITLE
[YUNIKORN-2296] Incorrect assert condition in config_test.go

### PR DESCRIPTION
**Pull request by brandboat**
### What is this PR for?
Fix incorrect assertion condition in config_test.go and add some missing test cases.

### What type of PR is it?
* [x] - Bug Fix

### Todos
N/A

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2296

### How should this be tested?
covered by unit tests

### Screenshots (if appropriate)
N/A

### Questions:
N/A

### CODE REVIEW

File: pkg/common/configs/config_test.go [hunk 1]
```diff
@@ -993,28 +993,28 @@ partitions:
 
 	// limit 1 check
 	limit := conf.Partitions[0].Queues[0].Limits[0]
-	if len(limit.Users) != 1 && limit.Users[0] != "user1" && len(limit.Groups) != 0 {
+	if len(limit.Users) != 1 || limit.Users[0] != "user1" || len(limit.Groups) != 0 {
 		t.Errorf("user and group list have incorrect entries: %v", limit)
 	}
-	if limit.MaxApplications != 5 && (len(limit.MaxResources) != 2 && limit.MaxResources["memory"] != "10000") {
+	if limit.MaxApplications != 5 || len(limit.MaxResources) != 2 || limit.MaxResources["memory"] != "10000" {
```
Comment (FrankYang0529) [inline pkg/common/configs/config_test.go:999]: Do we want to add one more condition `limit.MaxResources["vcore"] != "10"`?

File: pkg/common/configs/config_test.go [hunk 2]
```diff
@@ -916,35 +916,35 @@ partitions:
...
 	limit := conf.Partitions[0].Limits[0]
-	if len(limit.Users) != 1 && limit.Users[0] != "user1" && len(limit.Groups) != 0 {
+	if len(limit.Users) != 1 || limit.Users[0] != "user1" || len(limit.Groups) != 0 {
 		t.Errorf("failed to load max apps %v", limit)
 	}
 	if len(limit.Groups) != 0 {
 		t.Errorf("group list should be empty and is not: %v", limit)
 	}
-	if limit.MaxApplications != 10 && (len(limit.MaxResources) != 2 && limit.MaxResources["memory"] != "10000") {
+	if limit.MaxApplications != 5 || len(limit.MaxResources) != 2 || limit.MaxResources["memory"] != "10000" {
```
Comment (FrankYang0529) [inline pkg/common/configs/config_test.go:925]: Do we want to add one more condition `limit.MaxResources["vcore"] != "10"`?

Comment (brandboat) [reply pkg/common/configs/config_test.go]: Thanks for the comments !

**Review by FrankYang0529 — APPROVED** [review]
LGTM. Thanks for the contribution.

**Review by HuangTing-Yao — APPROVED** [review]
LGTM

**Pull request merged**